inject/src/main/java/io/micronaut/inject/BeanDefinition.java

```diff
@@ -267,7 +267,7 @@ default <R> Stream<ExecutableMethod<T, R>> findPossibleMethods(String name) {
     /**
      * @return The {@link ExecutableMethod} instances for this definition
      */
-    default Collection<ExecutableMethod<T, ?>> getExecutableMethods() {
+    default Collection<ExecutableMethod<T, Object>> getExecutableMethods() {
```

Comment:
This signature makes more sense and allows us to aggregate methods without a compiler error

Reply:
`?` is correct though? it would be better to change downstream users of `getExecutableMethods` to use proper covariance

tbh maybe we could just drop the return type parameter from ExecutableMethod, i dont think we use it productively anyway